### PR TITLE
Use the Vault ldaputil SDK

### LINKED ISSUE #7: LDAP
I noticed that this plugin is using its own LDAP utilities that were copied from Vault at the time of writing, which was definitely the way to go then. We've now made it possible to use the Vault SDK instead, which has updates and bug fixes.

Need to change to using that.

### PR DESCRIPTION
**Pull request by tyrannosaurus-becks**
Closes #7 

This PR switches over to using the Vault ldaputil SDK. This is useful because there have been some bug fixes since the copied code in this repo was added. It will also allow any further bug fixes to be pulled in when the SDK version is updated.

### CODE REVIEW

File: path_config_ldap.go [hunk 1]
```diff
@@ -127,41 +22,20 @@ Default: cn`,
...
-	// Create a new ConfigEntry, filling in defaults where appropriate
-	result, err := b.newConfigEntry(fd)
-	if err != nil {
-		return nil, err
-	}
-
-	storedConfig, err := req.Storage.Get(ctx, "config/ldap")
+// ConfigLDAP reads the present ldap config.
+func (b *backend) ConfigLdap(ctx context.Context, req *logical.Request) (*ldaputil.ConfigEntry, error) {
+	entry, err := req.Storage.Get(ctx, "config/ldap")
```
Comment (kalafut) [inline path_config_ldap.go]: Since `config/ldap` shows up in a few places it would be best to make a `const`.

File: go.mod [hunk 2]
```diff
@@ -10,13 +10,12 @@ require (
 	github.com/containerd/continuity v0.0.0-20190426062206-aaeac12a7ffc // indirect
 	github.com/docker/go-connections v0.4.0 // indirect
 	github.com/docker/go-units v0.4.0 // indirect
-	github.com/fatih/structs v1.1.0
-	github.com/go-ldap/ldap v3.0.3+incompatible
-	github.com/hashicorp/go-hclog v0.9.2
-	github.com/hashicorp/go-multierror v1.0.0
+	github.com/gotestyourself/gotestyourself v2.2.0+incompatible // indirect
+	github.com/hashicorp/go-hclog v0.8.0
```
Comment (kalafut) [inline go.mod]: Why the go-hclog downgrade?

**Comment by kalafut** [comment]
Just a suggestion and question, otherwise LGTM!

File: go.mod (hunk 2 above)
Comment (tyrannosaurus-becks) [reply go.mod]: I'm not sure why go mod chose that when I was updating the dependencies. Perhaps, given the other removed dependencies, this is now the minimum one that satisfies how it's used in the code? Seems like the main difference between those versions is the addition of context, but just to be safe, I can revert that change.

Comment (kalafut) [reply go.mod]: It looks list something in `vendor` was referencing 0.8.0, but it does all seem to build with 0.9.2.

**Pull request merged**